lib/wibox/container/radialprogressbar.lua

```diff
@@ -0,0 +1,264 @@
+        self._private.paddings[v] = val
+        self:emit_signal("widget::redraw_needed")
+    end
+end
+
+--- The padding between the outline and the progressbar.
+--@DOC_wibox_container_radialprogressbar_padding_EXAMPLE@
+-- @property paddings
+-- @tparam[opt=0] table|number paddings A number or a table
+-- @tparam[opt=0] number paddings.top
+-- @tparam[opt=0] number paddings.bottom
+-- @tparam[opt=0] number paddings.left
+-- @tparam[opt=0] number paddings.right
+
+--- The progressbar value.
+--@DOC_wibox_container_radialprogressbar_value_EXAMPLE@
+-- @property value
+-- @tparam number value Between min_value and max_value
+
+function radialprogressbar:set_value(val)
+    if not val then self._percent = 0; return end
+
+    if val > self._private.max_value then
+        self:set_max_value(val)
+    elseif val < self._private.min_value then
+        self:set_min_value(val)
+    end
+
+    local delta = self._private.max_value - self._private.min_value
+
+    self._percent = val/delta
+    self:emit_signal("widget::redraw_needed")
+end
+
+--- The border background color.
+--@DOC_wibox_container_radialprogressbar_border_color_EXAMPLE@
+-- @property border_color
+
+--- The border foreground color.
+--@DOC_wibox_container_radialprogressbar_color_EXAMPLE@
+-- @property color
+
+--- The border width.
+--@DOC_wibox_container_radialprogressbar_border_width_EXAMPLE@
+-- @property border_width
+-- @tparam[opt=3] number border_width
+
+--- The minimum value.
+-- @property min_value
+
+--- The maximum value.
+-- @property max_value
+
+for _, prop in ipairs {"max_value", "min_value", "border_color", "color",
+    "border_width", "paddings"} do
+    radialprogressbar["set_"..prop] = function(self, value)
+        self._private[prop] = value
+        self:emit_signal("property::"..prop)
+        self:emit_signal("widget::redraw_needed")
+    end
+    radialprogressbar["get_"..prop] = function(self)
+        return self._private[prop] or beautiful["radialprogressbar_"..prop]
+    end
+end
+
+function radialprogressbar:set_paddings(val)
+    self._private.paddings = type(val) == "number" and {
+        left   = val,
+        right  = val,
+        top    = val,
+        bottom = val,
+    } or val or {}
+    self:emit_signal("property::paddings")
+    self:emit_signal("widget::redraw_needed")
+    self:emit_signal("widget::layout_changed")
+end
+
+--- Returns a new radialprogressbar layout. A radialprogressbar layout
+-- radialprogressbars a given widget. Use `.widget` to set the widget.
+-- @param[opt] widget The widget to display.
+-- @function wibox.container.radialprogressbar
+local function new(widget)
+    local ret = base.make_widget(nil, nil, {
+        enable_properties = true,
+    })
+
+    util.table.crush(ret, radialprogressbar)
+    ret._private.max_value = 1
+    ret._private.min_value = 0
+
+    ret:set_widget(widget)
+
+    return ret
+end
+
+function radialprogressbar.mt:__call(...)
```

Comment:
This should be `_, ...` instead of `...` since the first argument is just `wibox.container.radialprogressbar`